File: everyshowing/src/main/java/com/wghcwc/everyshowing/ScreenUtils.java
package com.wghcwc.everyshowing;

import android.util.TypedValue;

import com.wghcwc.activitylifecycle2.ActivityLifecycle;

/**
 * @author wghcwc
 * @date 20-1-5
 */
public class ScreenUtils {

    public static int dp2px(float dpValue) {
        return (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dpValue, ActivityLifecycle.app.getResources().getDisplayMetrics());
    }
    /**
     * sp转px
     */
    public static int sp2px(float spValue) {
        return (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_SP, spValue, ActivityLifecycle.app.getResources().getDisplayMetrics());
    }
}

File: everyshowing/src/main/java/com/wghcwc/everyshowing/ScreenUtils.kt
package com.wghcwc.everyshowing

import android.util.TypedValue
import com.wghcwc.activitylifecycle2.ActivityLifecycle

/**
 * @author wghcwc
 * @date 20-1-5
 */
object ScreenUtils {
    @JvmStatic
    fun dp2px(dpValue: Float): Int {
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dpValue, ActivityLifecycle.getApp()?.resources?.displayMetrics).toInt()
    }

    /**
     * sp转px
     */
    @JvmStatic
    fun sp2px(spValue: Float): Int {
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_SP, spValue, ActivityLifecycle.getApp()?.resources?.displayMetrics).toInt()
    }
}